tests/test_symbols.py

```diff
@@ -595,6 +596,60 @@ def generate_symbols(self):
         yield A
         yield B
 
+    def test_truthy(self):
+        model = Model()
+
+        self.assertTrue(model.constant(1))
+        self.assertFalse(model.constant(0))
+        self.assertTrue(model.constant(1.1))
+        self.assertFalse(model.constant(0.0))
+
+        self.assertTrue(not model.constant(0))
+        self.assertFalse(not model.constant(1))
+
+        # these are all ambiguous
+        with self.assertRaises(ValueError):
+            bool(model.constant([]))
+        with self.assertRaises(ValueError):
+            bool(model.constant([0, 1]))
+        with self.assertRaises(ValueError):
+            bool(model.constant([0]))
+
+        # the type is correct
+        self.assertIsInstance(model.constant(123.4).__bool__(), bool)
+
+    def test_comparisons(self):
+        model = Model()
+
+        # zero = model.constant(0)
+        one = model.constant(1)
+        onetwo = model.constant([1, 2])
+
+        operators = [
+            operator.eq,
+            operator.ge,
+            operator.gt,
+            operator.le,
+            operator.lt,
+            operator.ne
+        ]
+
+        for op in operators:
```

Comment:
(minor) It would've been nice to have this parameterized and test each operator separately.

Reply:
Oh, true. I somehow missed that line, and running the test using pytest only shows this as a single test.

Reply:
[pytest-subtests](https://github.com/pytest-dev/pytest-subtests) required to make pytest work with `subTest`. Good to know!